File: Medium/2026-08-12-2958-Length-of-Longest-Subarray-With-at-Most-K-Frequency/solution.js
var longestSubarray = function(nums, k) {
    let left = 0, maxLength = 0;
    const frequency = new Map();

    for (let right = 0; right < nums.length; right++) {
        frequency.set(nums[right], (frequency.get(nums[right]) || 0) + 1);

        while (frequency.get(nums[right]) > k) {
            frequency.set(nums[left], frequency.get(nums[left]) - 1);
            if (frequency.get(nums[left]) === 0) {
                frequency.delete(nums[left]);
            }
            left++;
        }

        maxLength = Math.max(maxLength, right - left + 1);
    }

    return maxLength;
};

File: Medium/2026-08-12-2958-Length-of-Longest-Subarray-With-at-Most-K-Frequency/solution.java
import java.util.HashMap;

public class Solution {
    public int longestSubarray(int[] nums, int k) {
        HashMap<Integer, Integer> frequencyMap = new HashMap<>();
        int left = 0, maxLength = 0;

        for (int right = 0; right < nums.length; right++) {
            frequencyMap.put(nums[right], frequencyMap.getOrDefault(nums[right], 0) + 1);

            while (frequencyMap.get(nums[right]) > k) {
                frequencyMap.put(nums[left], frequencyMap.get(nums[left]) - 1);
                if (frequencyMap.get(nums[left]) == 0) {
                    frequencyMap.remove(nums[left]);
                }
                left++;
            }

            maxLength = Math.max(maxLength, right - left + 1);
        }

        return maxLength;
    }
}